### PR TITLE
Implement create account feature

### PR DESCRIPTION
**Pull request by hothanhtuan1209**
#17 
In this pull request, I perform the function of adding new accounts and adding user personal information
image
image
image
image

### CODE REVIEW

File: myproject/supermarket_manager/constants.py [hunk 1]
```diff
@@ -27,3 +27,15 @@
 
 # Message for assigned successfully
 ASSIGN = "Assign successfully"
+
+# Message not found role when add account
+NOT_FOUND_ROLE = "Role not found"
+
+# Message login_name or email exists
+LG_MAIL_EXISTS = "Login name or email already exists"
```
Comment (cuongdoan-agilityio) [inline myproject/supermarket_manager/constants.py:35]: Should update the constant name.

File: myproject/supermarket_manager/views.py [hunk 2]
```diff
@@ -261,3 +269,125 @@ def assign_permission(request):
...
+    
+    Returns:
+        JsonResponse: A JSON response indicating a result of the add operation
+    """
+
+
+    if request.method != "POST":
+        return JsonResponse({"message": INVALID_METHOD}, status=405)
+
+    else:
```
Comment (cuongdoan-agilityio) [inline myproject/supermarket_manager/views.py:301]: No need `else`.

File: myproject/supermarket_manager/models.py [hunk 3]
```diff
@@ -87,6 +82,22 @@ class Account(models.Model):
     gender = models.CharField(max_length=1, choices=GENDER_CHOICES)
```
[the remark this answers is not in the document]
Comment (hothanhtuan1209) [reply myproject/supermarket_manager/models.py:82]: Done

File: myproject/supermarket_manager/models.py [hunk 4]
```diff
@@ -87,6 +82,22 @@ class Account(models.Model):
...
     status = models.CharField(max_length=1, choices=STATUS_CHOICES, default="A")
 
+    def random_account_id(self):
```
[the remark this answers is not in the document]
Comment (hothanhtuan1209) [reply myproject/supermarket_manager/models.py]: Done

File: myproject/supermarket_manager/views.py (hunk 2 above)
Comment (hothanhtuan1209) [reply myproject/supermarket_manager/views.py]: Done

File: myproject/supermarket_manager/models.py [hunk 5]
```diff
@@ -87,6 +87,16 @@ class Account(models.Model):
...
 
+    def __str__(self):
+        """
+        Return a string representation of the Account object.
+        """
+
+        if isinstance(self.user_name, str):
+            return self.user_name
+        else:
+            return str(self.user_name)
```
[the remark this answers is not in the document]
Comment (hothanhtuan1209) [reply myproject/supermarket_manager/models.py]: Done

File: myproject/supermarket_manager/constants.py (hunk 1 above)
Comment (hothanhtuan1209) [reply myproject/supermarket_manager/constants.py]: Done

File: myproject/supermarket_manager/models.py [hunk 6]
```diff
@@ -71,9 +68,10 @@ class Account(models.Model):
         status (CharField): The status of the account.
     """
 
-    account_id = models.CharField(primary_key=True, max_length=5)
+    account_id = models.CharField(
```
Comment (cuongdoan-agilityio) [inline myproject/supermarket_manager/models.py:71]: Does not match with docstring.

File: myproject/supermarket_manager/views.py [hunk 7]
```diff
@@ -1,15 +1,21 @@
 from django.http import JsonResponse
 from django.db.utils import IntegrityError
 from django.views.decorators.csrf import csrf_exempt
-from .models import (Role, Permission, Role_Permission)
+from .models import (Role, Permission, Role_Permission, Account)
+import re
+from django.contrib.auth.hashers import make_password
+from django.contrib.auth import authenticate, login
```
Comment (cuongdoan-agilityio) [inline myproject/supermarket_manager/views.py:7]: I haven't seen you use these modules yet.

File: myproject/supermarket_manager/models.py (hunk 6 above)
Comment (hothanhtuan1209) [reply myproject/supermarket_manager/models.py:71]: Done

File: myproject/supermarket_manager/views.py (hunk 7 above)
Comment (hothanhtuan1209) [reply myproject/supermarket_manager/views.py]: Done

**Pull request merged**